### PR TITLE
[PWA] Add event page

### PR DESCRIPTION
**Pull request by tervay**
Adds event page with match results + alliance table page to new PWA. Not much else.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

File: pwa/app/components/tba/allianceSelectionTable.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import { Link } from '@remix-run/react';
+import { type VariantProps, cva } from 'class-variance-authority';
+import { max } from 'lodash-es';
```
Comment (fangeugene) [inline pwa/app/components/tba/allianceSelectionTable.tsx:3]: Can't we just use `Math.max`?

Comment (tervay) [reply pwa/app/components/tba/allianceSelectionTable.tsx:3]: That's fine, I typically find other use cases for lodash but can remove it here

File: pwa/package.json
Comment (fangeugene) [inline pwa/package.json:1]: 💀

File: pwa/app/routes/event.$eventKey.tsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+              Week {event.week + 1}
+            </Badge>
+          )}
+        </InlineIcon>
+
+        <InlineIcon>
+          <BiPinMapFill />
+
+          {event.gmaps_url ? (
+            <Link to={event.gmaps_url}>
```
Comment (fangeugene) [inline pwa/app/routes/event.$eventKey.tsx:69]: All these links on prod right now are `target="_blank" rel="noreferrer"`. should we do the same

File: pwa/app/routes/event.$eventKey.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import BiLink from '~icons/bi/link';
+import BiListOl from '~icons/bi/list-ol';
+import BiPinMapFill from '~icons/bi/pin-map-fill';
+import BiTrophy from '~icons/bi/trophy';
+import MdiFolderMediaOutline from '~icons/mdi/folder-media-outline';
+import MdiGraphBoxOutline from '~icons/mdi/graph-box-outline';
+import MdiRobot from '~icons/mdi/robot';
+import MdiTournament from '~icons/mdi/tournament';
+
+export async function loader({ params }: LoaderFunctionArgs) {
```
Comment (fangeugene) [inline pwa/app/routes/event.$eventKey.tsx:21]: Are we going to deal with the "skip the hop" stuff later?

File: pwa/app/routes/event.$eventKey.tsx (hunk 2 above)
Comment (tervay) [reply pwa/app/routes/event.$eventKey.tsx:69]: Up to you, I'm indifferent

File: pwa/app/routes/event.$eventKey.tsx (hunk 3 above)
Comment (tervay) [reply pwa/app/routes/event.$eventKey.tsx:21]: yes

File: pwa/app/routes/event.$eventKey.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    <>
+      <div className="mb-2.5 flex flex-col">
+        <h1 className="mb-2.5 mt-5 text-4xl">
+          {event.name} {event.year}
+        </h1>
+
+        <InlineIcon>
+          <BiCalendar />
+          {startDateStr} to {endDateStr}
+          {event.week && (
```
Comment (fangeugene) [inline pwa/app/routes/event.$eventKey.tsx:58]: `event.week` evaluates to `false` if week is zero. Should check for null/undefined instead.

File: pwa/app/routes/event.$eventKey.tsx (hunk 2 above)
Comment (fangeugene) [reply pwa/app/routes/event.$eventKey.tsx:69]: we can deal with that and other minor style/formatting stuff later.

File: pwa/app/routes/event.$eventKey.tsx (hunk 4 above)
Comment (tervay) [reply pwa/app/routes/event.$eventKey.tsx:58]: i hate everything

File: pwa/app/routes/event.$eventKey.tsx [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+export async function loader({ params }: LoaderFunctionArgs) {
+  if (params.eventKey === undefined) {
+    throw new Error('Missing eventKey');
+  }
+
+  const event = await getEvent({ eventKey: params.eventKey });
+  const matches = await getEventMatches({ eventKey: params.eventKey });
+  const alliances = await getEventAlliances({ eventKey: params.eventKey });
+
+  return json({ event, matches, alliances });
```
Comment (fangeugene) [inline pwa/app/routes/event.$eventKey.tsx:30]: How do we handle 404's if the event key provided is not a real event key?

Comment (tervay) [reply pwa/app/routes/event.$eventKey.tsx:30]: We can get to that later imo

**Pull request merged**